src/CanvasRenderingContext2d.cc

```diff
@@ -786,6 +787,28 @@ Context2d::SetDirection(const Napi::CallbackInfo& info, const Napi::Value& value
   state->direction = dir;
 }
 
+/*
+ * Get language.
+ */
+Napi::Value
+Context2d::GetLanguage(const Napi::CallbackInfo& info) {
+  return Napi::String::New(env, state->lang);
+}
+
+/*
+ * Set language.
+ */
+void
+Context2d::SetLanguage(const Napi::CallbackInfo& info, const Napi::Value& value) {
+  if (!value.IsString()) return;
+
+  std::string lang = value.As<Napi::String>();
+  state->lang = lang;
+
+  pango_context_set_language(pango_layout_get_context(_layout), pango_language_from_string(lang.c_str()));
```

Comment:
This should be done lazily too. Can you put it in `MeasureText` and `paintText`, right before `pango_cairo_update_layout`?